Firestore/Source/API/FIRQuerySnapshot.m

```diff
@@ -76,6 +77,35 @@ - (instancetype)initWithFirestore:(FIRFirestore *)firestore
   return self;
 }
 
+// NSObject Methods
+- (BOOL)isEqual:(nullable id)other {
+  if (other == self) return YES;
+  if (![[other class] isEqual:[self class]]) return NO;
+
+  return [self isEqualToSnapshot:other];
+}
+
+- (BOOL)isEqualToSnapshot:(nullable FIRQuerySnapshot *)snapshot {
+  if (self == snapshot) return YES;
+  if (snapshot == nil) return NO;
+  if (self.firestore != snapshot.firestore && ![self.firestore isEqual:snapshot.firestore])
```

Comment:
shouldn't these be `||`s ?

Reply:
No, should be &&. Check the case-analysis above (nil v.s. non-nil, non-nil v.s. non-nil). Another way to look at this is to use double-negation of it. Say, self.firestore is equal iff the reference is equal OR the content is equal. So self.firestore is not equal (case here to return NO) iff the reference is not equal AND the content is not equal.

Reply:
Er, sorry I keep trying to lead you astray.  For some reason when I was looking at this I was thinking of the equals case instead of the not-equals case. Carry on!